receiver/prometheusreceiver/internal/metricfamily.go

```diff
@@ -48,37 +41,32 @@ type metricFamily struct {
 	intervalStartTimeMs int64
 }
 
-func newMetricFamily(metricName string, mc MetadataCache, logger *zap.Logger, intervalStartTimeMs int64) MetricFamily {
-	familyName := normalizeMetricName(metricName)
-
-	// lookup metadata based on familyName
-	metadata, ok := mc.Metadata(familyName)
-	if !ok && metricName != familyName {
-		// use the original metricName as metricFamily
-		familyName = metricName
-		// perform a 2nd lookup with the original metric name. it can happen if there's a metric which is not histogram
-		// or summary, but ends with one of those _count/_sum suffixes
-		metadata, ok = mc.Metadata(metricName)
-		// still not found, this can happen when metric has no TYPE HINT
-		if !ok {
-			metadata.Metric = familyName
-			metadata.Type = textparse.MetricTypeUnknown
-		}
-	} else if !ok && isInternalMetric(metricName) {
-		metadata = defineInternalMetric(metricName, metadata, logger)
-	} else if !ok {
-		// Prometheus sends metrics without a type hint as gauges.
-		// MetricTypeUnknown is converted to a gauge in convToOCAMetricType()
-		metadata.Type = textparse.MetricTypeUnknown
+func metadataForMetric(metricName string, mc MetadataCache) (*scrape.MetricMetadata, string) {
+	if metadata, ok := internalMetricMetadata[metricName]; ok {
+		return metadata, metricName
+	}
+	if metadata, ok := mc.Metadata(metricName); ok {
+		return &metadata, metricName
+	}
+	// If we didn't find metadata with the original name,
+	// try with suffixes trimmed, in-case it is a "merged" metric type.
+	normalizedName := normalizeMetricName(metricName)
+	if metadata, ok := mc.Metadata(normalizedName); ok {
+		return &metadata, normalizedName
 	}
+	// Otherwise, the metric is unknown
+	return &scrape.MetricMetadata{
+		Metric: metricName,
+		Type:   textparse.MetricTypeUnknown,
+	}, metricName
+}
+
+func newMetricFamily(metricName string, mc MetadataCache, logger *zap.Logger, intervalStartTimeMs int64) *metricFamily {
+	metadata, familyName := metadataForMetric(metricName, mc)
 	ocaMetricType := convToOCAMetricType(metadata.Type)
 
-	// If a counter has a _total suffix but metadata is stored without it, keep _total suffix as the name otherwise
-	// the metric sent won't have the suffix
-	if ocaMetricType == metricspb.MetricDescriptor_CUMULATIVE_DOUBLE && strings.HasSuffix(metricName, metricSuffixTotal) {
```

Comment:
Now i'm trying to figure out what to do with this... If metric metadata is sent without a type suffix, it would seem strange to add it here, but that is what the previous behavior was.

Reply:
I'm keeping the current behavior for now, even though it seems strange to me.